src/main/webapp/dosave.jsp

```diff
@@ -5,368 +5,370 @@
-                                      + "[" + i + "]_entryid") + "';");
+                                        if (zieltabelle.equals("quelle_inedition") && zielattributArray[j].equals("EditionID")) {
+                                            ed = DBtoDB(request.getParameter(combinedFeldnamenArray[j] + "[" + i + "]"));
+                                        }
+                                    } else if (combinedFeldtypenArray[j].equals("checkbox")) {
+                                        sql += zielattributArray[j] + " = '" + (request.getParameter(combinedFeldnamenArray[j] + "[" + i + "]") != null && request.getParameter(combinedFeldnamenArray[j]
+                                                + "[" + i + "]").equals("on") ? "1" : "0") + "', ";
+                                    }
+                                }
+                                sql = sql.substring(0, sql.length() - 2);
+                                sql += " WHERE ID='" + request.getParameter(datenfeld.toLowerCase() + "[" + i + "]_entryid") + "';";
+                                String old_ed = "";
+                                if (!ed.equals("")) {
+                                    List<Map> editionLst = SaveHelper.getMappedList("select EditionID from quelle_inedition WHERE ID='" + request.getParameter(datenfeld.toLowerCase()
+                                            + "[" + i + "]_entryid") + "';");
 
-                                      if (editionLst.size() > 0) {
-                                          old_ed = editionLst.get(editionLst.size() - 1).get("EditionID").toString();
-                                      }
-                                  }
-                                  SaveHelper.insertOrUpdateSql(sql);
-                                  if (!ed.equals("")) {
-                                      SaveHelper.insertOrUpdateSql("Update ueberlieferung_edition set EditionID='" + ed + "' where EditionID=" + old_ed +
-                                        " and UeberlieferungID in (select h_u.ID from handschrift_ueberlieferung h_u, quelle_inedition q_i where q_i.ID=" +
-                                        request.getParameter(datenfeld.toLowerCase() + "[" + i + "]_entryid") + " and q_i.QuelleID=h_u.QuelleID)"
-                                      );
+                                    if (editionLst.size() > 0) {
+                                        old_ed = editionLst.get(editionLst.size() - 1).get("EditionID").toString();
+                                    }
+                                }
+                                SaveHelper.insertOrUpdateSql(sql);
+                                if (!ed.equals("")) {
+                                    SaveHelper.insertOrUpdateSql("Update ueberlieferung_edition set EditionID='" + ed + "' where EditionID=" + old_ed
```

Comment:
In an optimal case, all queries should be built by the SaveHelper, so the JSP template should not build any queries by itself